lib/src/scope.dart

```diff
@@ -46,9 +47,10 @@ class Scope {
   /// Calls an isolate-scoped RPC named [method] with [params].
   ///
   /// This always adds the `isolateId` parameter to the RPC.
-  Future<dynamic> sendRequest(String method, [Map<String, Object> params]) async {
+  Future<Map<String, dynamic>> sendRequest(String method,
+      [Map<String, Object> params]) async {
     var allParams = {"isolateId": isolateId}..addAll(params ?? {});
-    return await peer.sendRequest(method, allParams);
+    return DelegatingMap.typed/*<String, dynamic>*/(await peer.sendRequest(method, allParams));
```

Comment:
An `as` cast should be sufficient here. `JSON.decode` should instantiate all its maps as `Map<String, dynamic>`.